### PR TITLE
Allow for a more visually accurate progress bar.

### LINKED ISSUE #518: Allow more precise XP progress values

### PR DESCRIPTION
**Pull request by T14Raptor**
Resolves #518.

### CODE REVIEW

File: server/entity/experience.go [hunk 1]
```diff
@@ -108,21 +116,21 @@ func progressFromExperience(experience int) (level int, progress float64) {
...
-		return int(float64(level*level)*2.5 - 40.5*float64(level) + 360)
+		return float64(level*level)*2.5 - 40.5*float64(level) + 360
 	}
-	return int(float64(level*level)*4.5 - 162.5*float64(level) + 2220)
+	return float64(level*level)*4.5 - 162.5*float64(level) + 2220
 }
 
 // experienceForLevel returns the amount experience needed to reach level + 1.
-func experienceForLevel(level int) int {
+func experienceForLevel(level int) float64 {
```
Comment (Sandertv) [inline server/entity/experience.go:129]: There doesn't seem any reason to make this a float64.

File: server/entity/experience.go [hunk 2]
```diff
@@ -108,13 +116,13 @@ func progressFromExperience(experience int) (level int, progress float64) {
 }
 
 // experienceForLevels calculates the amount of experience needed in total to reach a certain level.
-func experienceForLevels(level int) int {
+func experienceForLevels(level int) float64 {
```
Comment (Sandertv) [inline server/entity/experience.go:119]: Would probably go back to returning an int here too. The less we return floats in these functions the better I think.

Comment (T14Raptor) [reply server/entity/experience.go:119]: Might lose some experience points by truncating here

Comment (Sandertv) [reply server/entity/experience.go:119]: Technically though experience can only be full numbers?

**Review by Sandertv — APPROVED** [review]
Looks good, thanks!

**Pull request merged**